File: app/src/main/java/java/android/notes/wrapper/fragments/core/NotesAdapter.java
package java.android.notes.wrapper.fragments.core;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import android.widget.ImageView;
import android.widget.TextView;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import java.android.notes.R;
import java.android.notes.core.Note;
import java.android.notes.listeners.IItemsClickListener;
import java.util.List;

public class NotesAdapter extends RecyclerView.Adapter<NotesAdapter.NoteViewHolder>{
    private IItemsClickListener listener;
    private List<Note> list;

    public void setList(List<Note> list) {
        this.list = list;
    }
    public void setListener(IItemsClickListener listener) {
        this.listener = listener;
    }

    @NonNull
    @Override
    public NoteViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_note,parent,false);

        return new NoteViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull NoteViewHolder holder, int position) {
        holder.getItemView().<TextView>findViewById(R.id.headLineToList).setText(list.get(position).getHeadline());
        holder.getItemView().<TextView>findViewById(R.id.dateToList).setText(list.get(position).getDate());
    }

    @Override
    public int getItemCount() {
        return list.size();
    }

    //////////////////////////////////////////////////////////////////////////////////////////////////////////
    public class NoteViewHolder extends RecyclerView.ViewHolder {
        private View itemView;

        public NoteViewHolder(@NonNull View itemView) {
            super(itemView);

            this.itemView = itemView;

            View.OnClickListener shortClick = new View.OnClickListener() {
                @Override
                public void onClick(View view) {

                    switch (view.getId()){
                        case R.id.noteInformation: listener.onOpenClick(getAdapterPosition()); break;
                        case R.id.imageViewDelete: listener.onDeleteClick(getAdapterPosition(),(ImageView)view); break;
                    }

                }
            };

            itemView.findViewById(R.id.noteInformation).setOnClickListener(shortClick);
            itemView.findViewById(R.id.imageViewDelete).setOnClickListener(shortClick);
            itemView.findViewById(R.id.noteInformation).setOnLongClickListener(new View.OnLongClickListener() {
                @Override
                public boolean onLongClick(View view) {
                    listener.onLongItemClick(view);
                    return true;
                }
            });

        }
        //////////////////////////////////////////////////////////////////////////////////////////////////////////

        public View getItemView() {
            return itemView;
        }
    }

}

File: app/src/main/java/java/android/notes/wrapper/fragments/core/NotesAdapter.kt
package java.android.notes.wrapper.fragments.core

import androidx.recyclerview.widget.RecyclerView
import java.android.notes.wrapper.fragments.core.NotesAdapter.NoteViewHolder
import java.android.notes.listeners.IItemsClickListener
import android.view.ViewGroup
import android.view.LayoutInflater
import android.view.View
import java.android.notes.R
import android.widget.TextView
import android.view.View.OnLongClickListener
import android.widget.ImageView
import java.android.notes.core.Note

class NotesAdapter() : RecyclerView.Adapter<NoteViewHolder>() {
    private var listener: IItemsClickListener? = null
    private var list: List<Note>? = null
    fun setList(list: List<Note>?) {
        this.list = list
    }

    fun setListener(listener: IItemsClickListener?) {
        this.listener = listener
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): NoteViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.item_note, parent, false)
        return NoteViewHolder(view)
    }

    override fun onBindViewHolder(holder: NoteViewHolder, position: Int) {
        holder.itemView.findViewById<TextView>(R.id.headLineToList).text =
            list!!.get(position).headline
        holder.itemView.findViewById<TextView>(R.id.dateToList).text = list!!.get(position).date
    }

    override fun getItemCount(): Int {
        return list!!.size
    }

    //////////////////////////////////////////////////////////////////////////////////////////////////////////
    inner class NoteViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        //////////////////////////////////////////////////////////////////////////////////////////////////////////

        init {

            //this.itemView = itemView
            val shortClick: View.OnClickListener = View.OnClickListener { view ->
                when (view.id) {
                    R.id.noteInformation -> listener!!.onOpenClick(adapterPosition)
                    R.id.imageViewDelete -> listener!!.onDeleteClick(
                        adapterPosition,
                        view as ImageView
                    )
                }
            }
            itemView.findViewById<View>(R.id.noteInformation).setOnClickListener(shortClick)
            itemView.findViewById<View>(R.id.imageViewDelete).setOnClickListener(shortClick)
            itemView.findViewById<View>(R.id.noteInformation).setOnLongClickListener(
                OnLongClickListener { view ->
                    listener!!.onLongItemClick(view)
                    true
                })
        }
    }
}